swap_meet/vendor.py

```diff
@@ -1,2 +1,86 @@
+from .item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    
+    def __init__(self, inventory = None):
+        self.inventory = inventory if inventory is not None else []
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        item_list = []
+        for item in self.inventory:
+            if item.category == category:
+                item_list.append(item)
+        return item_list
+
+    def swap_items(self, other, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other.inventory:
+            return False
+
+        self.remove(my_item)
+        other.add(my_item)
+        other.remove(their_item)
+        self.add(their_item)
+        return True
+
+    def swap_first_item(self, friend):
+        if len(self.inventory) == 0 or len(friend.inventory) == 0:
+            return False
+        
+        friend.add(self.inventory[0])
+        self.remove(self.inventory[0])
+        self.add(friend.inventory[0])
+        friend.remove(friend.inventory[0])
```

Comment:
This is essentially doing the same thing as `swap_items`, so we could call our function (as you did commented out below).

Though notice that due to the remove calls here or in `swap_items`, this will be O(n) where n is the lengths of the inventories, since we'll have to shift the whole remaining part of the inventories forward. By taking a different approach with `swap_items` or customizing our implementation here, we could do this swap in constant time.